addons/project/tests/test_project_sharing.py

```diff
@@ -323,16 +323,28 @@ def test_create_task_in_project_sharing(self):
             Task.create({'name': 'foo', 'child_ids': [Command.set([self.task_no_collabo.id])]})
 
         # Same thing but using context defaults
+        # However, cache is updated, but nothing is written.
         with self.assertRaisesRegex(AccessError, "top-secret records"):
             Task.with_context(default_child_ids=[Command.update(self.task_no_collabo.id, {'name': 'Foo'})]).create({'name': 'foo'})
+        with Task.env.cr.savepoint() as sp:
+            task = Task.with_context(default_child_ids=[Command.delete(self.task_no_collabo.id)]).create({'name': 'foo'})
+            task.env.invalidate_all()
+            self.assertTrue(self.task_no_collabo.exists(), "Task should still be there, no delete is sent")
+            sp.rollback()
+        with self.assertRaises(AccessError), self.env.cr.savepoint() as sp:
+            self.task_no_collabo.parent_id = self.task_no_collabo.create({'name': 'parent collabo'})
+            task = Task.with_context(default_child_ids=[Command.unlink(self.task_no_collabo.id)]).create({'name': 'foo'})
+            task.env.invalidate_all()  # raised here
+            self.assertTrue(self.task_no_collabo.parent_id, "Task should still be there, no delete is sent")
```

Comment:
Something is off here. The access error is raised on the line before and hence this line and the following are never called: The assert is not called and so useless, and the savepoint rollback doesn't happen

Reply:
The goal here (and in next lines) is to prevent having side-effects on other tests when the error is not raised, therefore rollback any changes if the line is reached.